Sources/Features/Breadcrumb/QueueFile.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+@objcMembers
+public class Queue<T>: NSObject {
+    private var elements: [T] = []
+
+    func enqueue(_ element: T) {
+        elements.append(element)
+    }
+
+    func dequeue() -> T? {
+        if elements.isEmpty {
+            return nil
+        } else {
+            return elements.removeFirst()
+        }
+    }
+
+    func peek() -> T? {
+        return elements.first
+    }
+
+    func pop() -> T? {
+        guard !elements.isEmpty else {
+                return nil
+            }
+        return elements.popLast()
+    }
+    
+    public func allElements() -> [T] {
```

Comment:
wouldn't it be better to create indexAt method instead of allElements?

Reply:
I can do that